src/test/ui/impl-trait/issues/issue-70877.stderr

```diff
@@ -2,13 +2,13 @@ error[E0271]: type mismatch resolving `<Bar as Iterator>::Item == Box<(dyn for<'
   --> $DIR/issue-70877.rs:9:12
    |
 LL | type FooRet = impl std::fmt::Debug;
-   |               -------------------- the expected opaque type
+   |               -------------------- the found opaque type
 ...
 LL | type Foo = impl Iterator<Item = FooItem>;
-   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected opaque type, found enum `Option`
+   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `Option`, found opaque type
    |
-   = note: expected struct `Box<(dyn for<'r> Fn(&'r (dyn ToString + 'r)) -> impl Debug + 'static)>`
-              found struct `Box<(dyn for<'r> Fn(&'r (dyn ToString + 'r)) -> Option<String> + 'static)>`
+   = note: expected struct `Box<(dyn for<'r> Fn(&'r (dyn ToString + 'r)) -> Option<String> + 'static)>`
+              found struct `Box<(dyn for<'r> Fn(&'r (dyn ToString + 'r)) -> impl Debug + 'static)>`
```

Comment:
This change looks like it might be wrong to me. On the other hand I'm not sure why [this test](https://github.com/rust-lang/rust/blob/b88e06c82084f81df5a730881d4f52f475d8abc7/src/test/ui/impl-trait/issues/issue-70877.rs) doesn't compile. `Bar`'s `Iterator` impl defines its `Item` to be `FooItem`, but the compiler complains `Bar` doesn't satisfy `impl Iterator<Item = FooItem>` because the `Item` type is wrong? It seems the compiler is being inconsistent about whether `ham` gets to see the true type of `FooRet`, or something.

At any rate this test is a complicated setup involving a nightly feature, so I don't think it's a big problem if this pr worsens the diagnostic in such a case.